pkg/util/kubernetes/apiserver/controllers.go

```diff
@@ -99,21 +112,44 @@ func startAutoscalersController(ctx ControllerContext) error {
 	if err != nil {
 		return err
 	}
+	informers := map[string]cache.SharedInformer{}
 	if ctx.WPAInformerFactory != nil {
 		go autoscalersController.RunWPA(ctx.StopCh, ctx.WPAClient, ctx.WPAInformerFactory)
+		informers["wpa"] = ctx.WPAInformerFactory.Datadoghq().V1alpha1().WatermarkPodAutoscalers().Informer()
 	}
 	// mutate the Autoscaler controller to embed an informer against the HPAs
 	autoscalersController.EnableHPA(ctx.InformerFactory.Autoscaling().V2beta1().HorizontalPodAutoscalers())
 	go autoscalersController.RunHPA(ctx.StopCh)
+	informers["hpa"] = ctx.InformerFactory.Autoscaling().V2beta1().HorizontalPodAutoscalers().Informer()
 
 	autoscalersController.RunControllerLoop(ctx.StopCh)
-	return nil
+
+	// Wait for the cache to sync
+	return SyncInformers(informers)
 }
 
+// startServicesInformer starts the service informer.
+// The synchronization of the service informer is handled in this function.
 func startServicesInformer(ctx ControllerContext) error {
 	// Just start the shared informer, the autodiscovery
 	// components will access it when needed.
 	go ctx.InformerFactory.Core().V1().Services().Informer().Run(ctx.StopCh)
 
-	return nil
+	// Wait for the cache to sync
+	return SyncInformers(map[string]cache.SharedInformer{
+		"services": ctx.InformerFactory.Core().V1().Endpoints().Informer(),
```

Comment:
```suggestion
		"services": ctx.InformerFactory.Core().V1().Services().Informer(),
```

Reply:
also, is it fine to do this rather than, for example:

```go
inf := map[string]cache.SharedInformer{}
svcInf := ctx.InformerFactory.Core().V1().Services().Informer()
inf["service"] = svcInf
go svcInf.Run(ctx.StopCh)
return SyncInformers(inf)
```
(short way of saying: can we call Run on one informer, and Sync on another? I guess that's okay because they are SharedInformer, but i'm not sure)

Reply:
thanks for noticing the Endpoints/Services mistake 🙇 
regarding the second comment, yes I believe it's fine, we already call the informers like this in AD https://github.com/DataDog/datadog-agent/blob/master/pkg/autodiscovery/listeners/kube_services.go#L61 without having to start them or something, it's shared.

Reply:
I can confirm that the cluster agent doesn't miss endpoint checks anymore with the current changes

Reply:
👌